handler/transport.go

```diff
@@ -37,24 +38,40 @@ type transport struct {
 	noAuthPaths []*policy.Assertion
 	// List to check for deprecated cipher suites
 	insecureCipherSuites []*tls.CipherSuite
+	metrics              service.Metrics
 }
 
 // Based on the following.
 // https://github.com/golang/oauth2/blob/bf48bf16ab8d622ce64ec6ce98d2c98f916b6303/transport.go
 func (t *transport) RoundTrip(r *http.Request) (*http.Response, error) {
+	var startTime time.Time
+
+	if t.metrics != nil {
+		defer func() {
+			if !startTime.IsZero() {
+				endTime := time.Since(startTime)
+				err := t.metrics.Observe(service.HTTP_ORIGIN_LATENCY, float64(endTime.Seconds()))
+				if err != nil {
+					glg.Errorf("cannot observe origin latency: %v", err)
```

Comment:
also, print path for debug
```suggestion
					glg.Errorf("cannot observe origin latency on: %s, err: %v", r.URL.Path, err)
```

Reply:
@WindzCUHK fixed in https://github.com/AthenZ/authorization-proxy/pull/30/commits/724c71a7ccdf857e4075e21b21d812c2868631f9